beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -344,42 +344,39 @@ func sortedBlockWithVerifiedBlobSlice(blocks []interfaces.ReadOnlySignedBeaconBl
 }
 
 func blobRequest(bwb []blocks2.BlockWithVerifiedBlobs, blobWindowStart primitives.Slot) *p2ppb.BlobSidecarsByRangeRequest {
-	if len(bwb) == 0 {
-		return nil
-	}
-	// Short-circuit if the highest block is before the deneb start epoch or retention period start.
-	// This assumes blocks are sorted by sortedBlockWithVerifiedBlobSlice.
-	highest := bwb[len(bwb)-1].Block.Block().Slot()
-	// bwb is sorted by slot, so if the last element is outside the retention window, no blobs are needed.
-	if highest < blobWindowStart {
-		return nil
-	}
 	lowest := lowestSlotNeedsBlob(blobWindowStart, bwb)
 	if lowest == nil {
 		return nil
 	}
+	highest := bwb[len(bwb)-1].Block.Block().Slot()
 	return &p2ppb.BlobSidecarsByRangeRequest{
 		StartSlot: *lowest,
 		Count:     uint64(highest.SubSlot(*lowest)) + 1,
 	}
 }
 
 func lowestSlotNeedsBlob(retentionStart primitives.Slot, bwb []blocks2.BlockWithVerifiedBlobs) *primitives.Slot {
-	i := sort.Search(len(bwb), func(i int) bool {
-		if bwb[i].Block.Block().Slot() < retentionStart {
-			return false
+	if len(bwb) == 0 {
+		return nil
+	}
+	// Short-circuit if the highest block is before the deneb start epoch or retention period start.
+	// This assumes blocks are sorted by sortedBlockWithVerifiedBlobSlice.
+	// bwb is sorted by slot, so if the last element is outside the retention window, no blobs are needed.
+	if bwb[len(bwb)-1].Block.Block().Slot() < retentionStart {
+		return nil
+	}
+	for _, b := range bwb {
+		slot := b.Block.Block().Slot()
+		if slot < retentionStart {
+			continue
 		}
-		commits, err := bwb[i].Block.Block().Body().BlobKzgCommitments()
+		commits, err := b.Block.Block().Body().BlobKzgCommitments()
```

Comment:
Do we really want to return `nil` here on errors? the caller will also return early with a `nil` when perhaps we want to propagate an error.

Reply:
Something I'm trying to do more often is minimize error handling to callers. In this case we know these accessors just return errors when a property isn't supported by the version. It isn't worth propagating that IMO.